docs/repositories-configure/codacy-configuration-file.md

```diff
@@ -17,7 +17,13 @@ Codacy supports configuring certain advanced features through a configuration fi
 ## Using a Codacy configuration file
 
 !!! important
-    If your repository has a Codacy configuration file, the [Ignored files settings](ignoring-files.md) defined on the Codacy UI don't apply and you must [ignore files using the configuration file](#syntax-for-ignoring-files) instead.
+    -   If your repository has a Codacy configuration file, the [Ignored files settings](ignoring-files.md) defined on the Codacy UI don't apply and you must [ignore files using the configuration file](#syntax-for-ignoring-files) instead.
+    -   Codacy uses the version of the configuration file **in the branch being analyzed**. For example, if you open a pull request that includes changes to the Codacy configuration file, the analysis results take those changes into account. However, if the Codacy configuration file **in the default branch** includes settings to [ignore files](#syntax-for-ignoring-files):
+
+        -   The analysis results of any branch always take the ignore settings defined **in the default branch** into account.
+        -   The ignore settings of the Codacy configuration file **in the branch being analyzed** are cumulative and don't override the settings defined **in the default branch**.
+        
+        For example, if the Codacy configuration file **in the default branch** excludes file A from analysis, that file will be excluded from analysis **in any branch being analyzed** even if you remove that rule from the Codacy configuration file in that branch.
```

Comment:
@nicklem I could use your help here! 🙏 
I believe the content is technically correct ([see the thread](https://codacy.slack.com/archives/C8X9SS1H7/p1712843138961309?thread_ts=1712766443.370299&cid=C8X9SS1H7)), but it seems a bit verbose. Another pair of expert eyes can probably make it simpler. 😅 👀

Reply:
This is a tough one, definitely! Here are a couple of ideas. We can discuss these later (or even during the offiste!)

```
- When a pull request modifies the Codacy configuration file, Codacy uses the updated configuration file to analyze the pull request. However, this does not apply to ignored files (see next point).
- If a pull request removes an ignored path from the Codacy configuration file, any matching files will stay ignored until the pull request is merged in the default branch.
```

Reply:
@nicklem Thank you for these much more clear examples! 👍 
I also split the admonition to avoid a huge wall of text and moved the ignore-related part to the corresponding section.
Let me know how it looks now! 😄